### PR TITLE
Fix x86 segm

### PR DESCRIPTION
**Pull request by serpilliere**
Sanitize/Fix some x86 semantic opcodes

### CODE REVIEW

File: test/arch/x86/unit/mn_pushpop.py [hunk 1]
```diff
@@ -117,12 +120,202 @@ def test_init(self):
...
+       JMP lbl_ret
+    '''
+
+    def check(self):
+        assert self.myjit.cpu.ESP == self.stk_origin + 0x4
+        assert self.myjit.cpu.EAX == self.value
+
+
+class Test_POP_mode32_16(Asm_Test_32):
+    MYSTRING = "test pop mode32 32"
```
Comment (commial) [inline test/arch/x86/unit/mn_pushpop.py]: It looks like it is "mode 32 16"

File: test/arch/x86/unit/mn_pushpop.py [hunk 2]
```diff
@@ -117,12 +120,202 @@ def test_init(self):
...
+       JMP lbl_ret
+    '''
+
+    def check(self):
+        assert self.myjit.cpu.ESP == self.stk_origin + 0x2
+        assert self.myjit.cpu.AX == self.value
+
+
+class Test_POP_mode16_32(Asm_Test_16):
+    MYSTRING = "test pop mode16 16"
```
Comment (commial) [inline test/arch/x86/unit/mn_pushpop.py]: And here, mode16 32

**Comment by commial** [comment]
Nice spot!

File: test/arch/x86/unit/mn_pushpop.py (hunk 1 above)
Comment (serpilliere) [inline test/arch/x86/unit/mn_pushpop.py]: Ho yes. Thanks

File: test/arch/x86/unit/mn_pushpop.py (hunk 2 above)
Comment (serpilliere) [inline test/arch/x86/unit/mn_pushpop.py]: Correct

**Pull request merged**